talk/owt/sdk/p2p/p2ppeerconnectionchannel.cc

```diff
@@ -649,19 +657,15 @@ void P2PPeerConnectionChannel::OnDataChannel(
   // Currently only one data channel for one connection. If we are going to
   // support multiple data channels(one for text, one for large files), replace
   // |data_channel_| with a map.
-  if (data_channel_)
-    data_channel_ = nullptr;
   data_channel_ = data_channel;
   data_channel_->RegisterObserver(this);
-  DrainPendingMessages();
```

Comment:
why was this removed?

Reply:
It's redundant with OnDataChannelStateChange(kOpen). Same reason why we removed CheckWaitedList() from OnIceConnectionChange(kConnected)